data/transactions/transaction.go

```diff
@@ -291,6 +293,15 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 			}
 		}
 
+		if proto.EnableBlockProofKeyregCheck {
```

Comment:
I think that you want to ensure that this `BlockProofPK` is set on online key registration transaction, and clear on offline key registration transaction. ( but I could be wrong here ).